src/dotnet-install.sh

```diff
@@ -587,6 +587,11 @@ validate_remote_local_file_sizes()
     fi 
 }
 
+# Function to convert a string to an integer
+function string_to_int {
+    echo "$1" | tr -d '\n' | sed 's/[^0-9]*//g'
```

Comment:
What does the problematic `remote_file_size` look like? If it's the trailing newline that's causing the issue, I believe it's fixable when the variable is assigned to.

Would this help in `get_remote_file_size`?
```bash
... | awk '{ORS=""; print $2}'
```

Reply:
The `ORS=""` part specifically.

Reply:
> The `ORS=""` part specifically.

`remote_file_size` looks like "123", when "file_size" is 123. 
Unfortunately, the suggested` ORS="" `didn't help. on `[ "$file_size" -ne "$remote_file_size" ]` it causes "integer expression expected", on `[ "$file_size" != "$remote_file_size" ]` returns invalid comparison

Reply:
Ah, so if we need to get rid of the quotes, then:

```bash
... | awk '{gsub("\"","",$2); print $2}'
```

Reply:
I used a bit different thing, please take a look

Reply:
Looks ok but I still think it should happen as part of the original assignment to remote_file_size in:

https://github.com/dotnet/install-scripts/blob/cab6fcd7d451b3284e45b6eff7f1fd8d324ece17/src/dotnet-install.sh#L964,L966

- It would be consistent with the variable holding the local size (both are integers from the get go).
- As a bonus it will run with one less process created (we already use `awk` there).

Reply:
Apologies for scrutinizing but it's important to understand what the exact issue is we're fixing. Apparently it's not newline and it's not quotes:

```
~$ curl -sI http://some.random.url | grep -i content-length | awk '{print $2}' | xxd
00000000: 3438 3836 3132 3830 0d0a                 48861280..
~$ curl -sI http://some.random.url | grep -i content-length | awk '{num = $2 + 0; print num}' | xxd
00000000: 3438 3836 3132 3830 0a
```

It's the CR (0d) character that throws the shell off because HTTP uses the CRLF "Windows-style" new-line sequence.